protocol/x/clob/e2e/short_term_orders_test.go

```diff
@@ -607,3 +607,370 @@ func TestPlaceOrder(t *testing.T) {
+						},
+					},
+				},
+			},
+		},
+		"Success: Replace in next block": {
+			blocks: []blockOrdersAndExpectations{
+				{
+					ordersToPlace: []clobtypes.MsgPlaceOrder{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20,
+					},
+					orderIdsExpectations: map[clobtypes.OrderId]orderIdExpectations{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20.Order.OrderId: {
+							shouldExistOnMemclob: true,
+							expectedOrder:        PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20.Order,
+						},
+					},
+				},
+				{
+					ordersToPlace: []clobtypes.MsgPlaceOrder{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB21,
+					},
+					orderIdsExpectations: map[clobtypes.OrderId]orderIdExpectations{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB21.Order.OrderId: {
+							shouldExistOnMemclob: true,
+							expectedOrder:        PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB21.Order,
+						},
+					},
+				},
+			},
+		},
+		"Fail: Replacement order has lower GTB than existing order": {
+			blocks: []blockOrdersAndExpectations{
+				{
+					ordersToPlace: []clobtypes.MsgPlaceOrder{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB21,
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20,
+					},
+					orderIdsExpectations: map[clobtypes.OrderId]orderIdExpectations{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB21.Order.OrderId: {
+							shouldExistOnMemclob: true,
+							expectedOrder:        PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB21.Order,
+						},
+					},
+				},
+			},
+		},
+		"Fail: Replacement order has equal GTB to existing order": {
+			blocks: []blockOrdersAndExpectations{
+				{
+					ordersToPlace: []clobtypes.MsgPlaceOrder{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20,
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB20,
+					},
+					orderIdsExpectations: map[clobtypes.OrderId]orderIdExpectations{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20.Order.OrderId: {
+							shouldExistOnMemclob: true,
+							expectedOrder:        PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20.Order,
+						},
+					},
+				},
+			},
+		},
+		"Success: Replacement order after partial match in same block": {
+			blocks: []blockOrdersAndExpectations{
+				{
+					ordersToPlace: []clobtypes.MsgPlaceOrder{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20,
+						*clobtypes.NewMsgPlaceOrder(MustScaleOrder(
+							clobtypes.Order{
+								OrderId:      clobtypes.OrderId{SubaccountId: constants.Bob_Num0, ClientId: 0, ClobPairId: 0},
+								Side:         clobtypes.Order_SIDE_SELL,
+								Quantums:     3,
+								Subticks:     10,
+								GoodTilOneof: &clobtypes.Order_GoodTilBlock{GoodTilBlock: 20},
+							},
+							testapp.DefaultGenesis(),
+						)),
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB21,
+					},
+					orderIdsExpectations: map[clobtypes.OrderId]orderIdExpectations{
```

Comment:
Can we add some expectations that `PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20` does not exist anywhere?

Also the 2nd message that was placed should have a fill amount too?

Reply:
As for 1) By asserting that `PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB21.Order` exists on the memclob, we are simultaneously asserting that `PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB20.Order` does not exist. These two orders share an OrderId and all data stored on both the memclob and in state is keyed off of OrderId. 

2). Yes the second order should have a fill amount but honestly I feel like it is fine to not assert that here. It makes the test simpler to read by only asserting information relevant to the order being replaced. We have other tests that validate matching